extensions/data-plane/data-plane-s3/src/main/java/org/eclipse/dataspaceconnector/aws/dataplane/s3/S3DataSink.java

```diff
@@ -20,28 +20,65 @@
 import org.eclipse.dataspaceconnector.spi.response.StatusResult;
 import software.amazon.awssdk.core.sync.RequestBody;
 import software.amazon.awssdk.services.s3.S3Client;
-import software.amazon.awssdk.services.s3.model.PutObjectRequest;
+import software.amazon.awssdk.services.s3.model.CompleteMultipartUploadRequest;
+import software.amazon.awssdk.services.s3.model.CompletedMultipartUpload;
+import software.amazon.awssdk.services.s3.model.CompletedPart;
+import software.amazon.awssdk.services.s3.model.CreateMultipartUploadRequest;
+import software.amazon.awssdk.services.s3.model.UploadPartRequest;
 
 import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.util.ArrayList;
 import java.util.List;
 
 class S3DataSink extends ParallelSink {
 
     private S3Client client;
     private String bucketName;
     private String keyName;
+    private int chunkSize;
 
-    private S3DataSink() { }
+    private S3DataSink() {}
 
     @Override
     protected StatusResult<Void> transferParts(List<DataSource.Part> parts) {
         for (DataSource.Part part : parts) {
             try (var input = part.openStream()) {
-                PutObjectRequest request = PutObjectRequest.builder()
+
+                int partNumber = 1;
+                List<CompletedPart> completedParts = new ArrayList<>();
+
+                String uploadId = client.createMultipartUpload(CreateMultipartUploadRequest.builder()
+                        .bucket(bucketName)
+                        .key(keyName)
+                        .build()).uploadId();
+
+                while (true) {
```

Comment:
I know that there aren't many way to implement this and this implementation is correct, but if it would be possible to avoid the `while (true)` it would be great. It's just a personal taste

Reply:
I've tried, I've really tried :) As checkstyle forbids inline assignments, avoiding while (true) in a reasonable way would be... difficult?  :)

Reply:
I'm ok to leave it like it is